messaging/handler_test.go

```diff
@@ -182,6 +183,91 @@ func TestHandler_ErrNotFound(t *testing.T) {
 	}
 }
 
+// Ensure a handler can correctly create a replica.
+func TestHandler_createReplica(t *testing.T) {
+	s := NewServer()
+	defer s.Close()
+
+	// Send request to the broker.
+	resp, _ := http.Post(s.URL+`/messaging/replicas?id=200`, "application/octet-stream", nil)
+	defer resp.Body.Close()
+	if resp.StatusCode != http.StatusCreated {
+		t.Fatalf("unexpected status: %d", resp.StatusCode)
+	}
+
+	// Verify replica was created.
+	if r := s.Handler.Broker().Replica(200); r == nil {
+		t.Fatalf("replica not created")
+	}
+}
+
+// Ensure a handler returns an error when creating a replica without an id.
+func TestHandler_createReplica_ErrReplicaIDRequired(t *testing.T) {
+	s := NewServer()
+	defer s.Close()
+
+	// Send request to the broker.
+	resp, _ := http.Post(s.URL+`/messaging/replicas`, "application/octet-stream", nil)
+	defer resp.Body.Close()
+	if resp.StatusCode != http.StatusBadRequest {
+		t.Fatalf("unexpected status: %d", resp.StatusCode)
+	} else if resp.Header.Get("X-Broker-Error") != "replica id required" {
+		t.Fatalf("unexpected error: %s", resp.Header.Get("X-Broker-Error"))
+	}
+}
+
+// Ensure a handler returns an error when creating a replica that already exists.
+func TestHandler_createReplica_ErrReplicaExists(t *testing.T) {
+	s := NewServer()
+	defer s.Close()
+	s.Handler.Broker().CreateReplica(200)
+
+	// Send request to the broker.
+	resp, _ := http.Post(s.URL+`/messaging/replicas?id=200`, "application/octet-stream", nil)
+	defer resp.Body.Close()
+	if resp.StatusCode != http.StatusConflict {
+		t.Fatalf("unexpected status: %d", resp.StatusCode)
+	} else if resp.Header.Get("X-Broker-Error") != "replica already exists" {
+		t.Fatalf("unexpected error: %s", resp.Header.Get("X-Broker-Error"))
+	}
+}
+
+// Ensure a handler can correctly delete a replica.
+func TestHandler_deleteReplica(t *testing.T) {
+	s := NewServer()
+	defer s.Close()
+	s.Handler.Broker().CreateReplica(200)
+
+	// Send request to the broker.
+	req, _ := http.NewRequest("DELETE", s.URL+`/messaging/replicas?id=200`, nil)
+	resp, _ := http.DefaultClient.Do(req)
+	defer resp.Body.Close()
+	if resp.StatusCode != http.StatusNoContent {
+		t.Fatalf("unexpected status: %d (%s)", resp.StatusCode, resp.Header.Get("X-Broker-Error"))
+	}
+
+	// Verify replica was deleted.
+	if r := s.Handler.Broker().Replica(200); r != nil {
+		t.Fatalf("replica not deleted")
+	}
+}
+
+// Ensure a handler returns an error when deleting a replica without an id.
```

Comment:
Perhaps test case ensuring that deleting a replica that doesn't exist gets the error you expect?